src/librustc_infer/infer/lexical_region_resolve/mod.rs

```diff
@@ -325,8 +325,16 @@ impl<'cx, 'tcx> LexicalResolver<'cx, 'tcx> {
             }
         }
 
-        debug!("enforce_member_constraint: final least choice = {:?}", least_choice);
-        if least_choice != member_lower_bound {
+        // Take the lub, so that in the case that the least choice and
+        // member_lower_bound are distinct `ty::Regions` which happen to be
+        // equal we don't change the vars value, only to have `expansion`
+        // change it back.
```

Comment:
maybe cc the related issue in this comment

Reply:
I'd like to understand just a bit better what the "related regions that happen to be equal" are

Reply:
I've clarified that this means universal regions that are known to be equal from (implied) bounds on the function.